autogen/main/cluster.tf.tmpl

```diff
@@ -248,6 +248,10 @@ resource "google_container_cluster" "primary" {
     config_connector_config {
       enabled = var.config_connector
     }
+
+    gke_backup_agent_config {
```

Comment:
Lets make this a dynamic block similar to `gce_persistent_disk_csi_driver_config` to prevent any issues like https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1255

Reply:
done